stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/CandidateResult.java

```diff
@@ -54,25 +99,69 @@ public String getDocId() {
         return docId;
     }
 
+    public String getDirectEntityName() {
+        return directEntityName;
+    }
+
+    public UUID getDirectEntityUUID() {
+        return directEntityUUID;
+    }
+
+    public String getDirectEntityType() {
+        return directEntityType;
+    }
+
+    // use to set id for direct query after resolution
+    public void setId(Id entityId) {
+        this.entityId = entityId;
+    }
+
 
     @Override
     public boolean equals( final Object o ) {
         if ( this == o ) {
             return true;
         }
+        if ( o == null ) {
+            return false;
+        }
         if ( !( o instanceof CandidateResult ) ) {
             return false;
         }
 
         final CandidateResult that = ( CandidateResult ) o;
 
-        if ( !entityId.equals( that.entityId ) ) {
+        if ( entityId == null && that.entityId != null) {
```

Comment:
Do we have extra tests for all these additional conditions to equals()

Reply:
I'm guessing you looked and there weren't -- Keyur, please add.